app/Http/Requests/UpsertFlightRequest.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace App\Http\Requests;
+
+use Illuminate\Foundation\Http\FormRequest;
+
+class UpsertFlightRequest extends FormRequest
+{
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, \Illuminate\Contracts\Validation\ValidationRule|array|string>
+     */
+    public function rules(): array
+    {
+        return [
+            'airline_id' => ['required', 'integer', 'exists:airlines,id'],
+            'origin_city_id' => ['required', 'integer', 'exists:cities,id'],
+            'destination_city_id' => ['required', 'integer', 'exists:cities,id', 'different:origin_city_id'],
+            'departure_at' => ['required', 'date_format:Y-m-d\TH:i'],
+            'arrival_at' => ['required', 'date', 'date_format:Y-m-d\TH:i', 'after:departure_at'],
```

Comment:
```suggestion
            'arrival_at' => ['required', 'date_format:Y-m-d\TH:i', 'after:departure_at'],
```
When using the `date_format` rule, you don't have to use the `date` rule, and viceversa.